### PR TITLE
NED Annotations for Quaero (Resolves #702)

### LINKED ISSUE #702: quaero: missing NED
From the corpus webpage: https://quaerofrenchmed.limsi.fr/
```
 T1 PROC 3 16 contraception
#1 AnnotatorNotes T1 C0700589
T2 DEVI 25 50 dispositifs intra utérins
#2 AnnotatorNotes T2 C0021900
T3 ANAT 43 50 utérins
#3 AnnotatorNotes T3 C0042149 
```
The `AnnotatorNotes` are UMLS codes

### PR DESCRIPTION
**Pull request by phlobo**
Resolves #702 

- Annotations are read from BRAT files instead of BioC, because 1 document was missing in BioC. Numbers of documents / entities in unit test output match http://ceur-ws.org/Vol-1609/16090028.pdf
- NED annotations added
- Following discussion in https://github.com/bigscience-workshop/biomedical/issues/731, multiple CUIs per entities are currently *not* merged, as the vast majority differs by type and CUI
- Annotator Notes in BRAT can be empty, which caused an error in the parser

### CODE REVIEW

File: bigbio/biodatasets/quaero/quaero.py [hunk 1]
```diff
@@ -225,86 +180,39 @@ def _split_generators(self, dl_manager):
...
-                                    self._get_bioc_entity(span, doc_text)["offsets"][0]
-                                    for span in passage.annotations
-                                ],
-                            }
-                            for passage in xdoc.passages
-                        ],
-                    },
-                )
-                key += 1
+            for guid, txt_file in enumerate(folder.glob("*.txt")):
```
Comment (galtay) [inline bigbio/biodatasets/quaero/quaero.py:191]: I'm a little concerned that `glob` will not return a deterministic order for the files. Is there a sort key on the file names that makes sense?

File: bigbio/biodatasets/quaero/quaero.py [hunk 2]
```diff
@@ -225,86 +180,39 @@ def _split_generators(self, dl_manager):
...
-                for passage in xdoc.passages:
-                    for span in passage.annotations:
-                        ent = self._get_bioc_entity(span, doc_text)
-                        ent["id"] = uid  # override BioC default id
-                        data["entities"].append(ent)
-                        uid += 1
-
-                yield i, data
-                i += 1
+            for guid, txt_file in enumerate(folder.glob("*.txt")):
```
Comment (galtay) [inline bigbio/biodatasets/quaero/quaero.py:202]: same file ordering concern here

File: bigbio/biodatasets/quaero/quaero.py (hunk 1 above)
Comment (phlobo) [reply bigbio/biodatasets/quaero/quaero.py:191]: The file name seems to be some kind of document id, so sorting by file name would be an option.
To be honest, I just copied that part from [examples/mlee.py](https://github.com/bigscience-workshop/biomedical/blob/master/examples/mlee.py), as this was given as a reference in the contribution guide.

>BRAT: Use [our custom brat parser](https://github.com/bigscience-workshop/biomedical/blob/master/bigbio/utils/parsing.py). Example usage can be found in [examples/mlee.py](https://github.com/bigscience-workshop/biomedical/blob/master/examples/mlee.py).

Comment (galtay) [reply bigbio/biodatasets/quaero/quaero.py]: ok, looks good, we can update in another PR if there turns out to be a canonical order that we want to enforce.  
@leonweber @sg-wbi @WojciechKusa I think we should probably update that example ^

**Review by galtay — APPROVED** [review]
thanks @phlobo !

**Pull request merged**